Keen/AccessKey/AccessKey.cs

```diff
@@ -0,0 +1,79 @@
+﻿using Keen.Core.Query;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Newtonsoft.Json.Serialization;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Keen.Core.AccessKey
+{
+    /// <summary>
+    /// Modal for AccessKey object
+    /// </summary>
+    public class AccessKey
+    {
+        public string Name { get; set; }
+        public bool IsActive { get; set; }
+        public ISet<string> Permitted { get; set; }
+        public string Key { get; set; }
+        public Options Options { get; set; }
+
+        internal AccessKey(string name, bool isActive, ISet<string> permitted, Options options)
```

Comment:
If the properties all remain public, is it useful to have an internal constructor instead of using object initializer notation and giving the `Key` auto-prop a default?